proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,257 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
 
 In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the `field` into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
 
+## Detailed design
+
+For properties with an `init` accessor, everything that applies below to `set` would apply instead to the `init` accessor.
+
+**Principle 1:** Every property declaration can be thought of as having a backing field by default, which is elided when not used. The field is referenced using the keyword `field` and its visibility is scoped to the accessor bodies.
```

Comment:
What I was trying to get at in a separate comment thread, and will reiterate here is: I don't think this is a useful framing.

I think this framing implies that the compiler can simplify the way it handles properties by making an assumption that the backing field exists, until some particular phase where it needs to finally decide for sure, but I don't think this is true. It's not a "late-bound" decision, e.g. which we can wait until the lowering or emit layer to decide on it. It affects the results and meaning of initial binding, and the answers for whether a diagnostic should be reported when taking a pointer to a type, nullable analysis, the result of `GetMembers()` public API, and probably several other things.

I think it would be more useful here to be specific and say that the backing field exists or not based on whether a contextual keyword is being used in the property accessors. Basically, our design has accounted for the need for the compiler to make a decision about whether the field exists relatively early on.

Reply:
Are the Principles necessary? It looks like the update below to §14.7.4 Automatically implemented properties fully describes when the backing field is available. Consider removing the Principles to avoid an extra concept.

Reply:
The update to *§14.7.4 Automatically implemented properties* should be first.

Reply:
i like things like this because they guide intuition on how to think about the feature space as a whole.  Properties are simply wrappers around fields that can run code.  but the lang considers the field something that a supporting impl is free to elide.

---

In terms of *when* it elides, we can hten have another section describing that.  but it aligns with the principles laid out here.

Reply:
The principles help to organize the thinking around the proposal and future updates in this area, compared to inferring from the concrete individual changes. I would really like to have them be considered (and potentially, rejected) at a philosophical level, which then is an inherent part of the design.

Reply:
I have removed the framing of "every property has a backing field" (restated as "every property can make use of a backing field"), and the framing of "get;" and "set;" being syntactic sugar over `=> field` and `=> field = value;`, which guided our design. I'd like to come back to them, but I don't want to overload the focus on the other open questions for now.